src/plugins/NEMD/MettDeamon.cpp

```diff
@@ -1584,25 +1678,25 @@ void Reservoir::sortParticlesToBins()
 	cout << "["<<ownRank<<"]: _particleVector.size()=" << _particleVector.size() << endl;
 
 	cout << "["<<ownRank<<"]: bbMin ="
-			<< domainDecomp.getBoundingBoxMin(0, domain) << ", "
-			<< domainDecomp.getBoundingBoxMin(1, domain) << ", "
-			<< domainDecomp.getBoundingBoxMin(2, domain) << "; bbMax = "
-			<< domainDecomp.getBoundingBoxMax(0, domain) << ", "
-			<< domainDecomp.getBoundingBoxMax(1, domain) << ", "
-			<< domainDecomp.getBoundingBoxMax(2, domain) << endl;
+			<< domainDecomp->getBoundingBoxMin(0, domain) << ", "
+			<< domainDecomp->getBoundingBoxMin(1, domain) << ", "
+			<< domainDecomp->getBoundingBoxMin(2, domain) << "; bbMax = "
+			<< domainDecomp->getBoundingBoxMax(0, domain) << ", "
+			<< domainDecomp->getBoundingBoxMax(1, domain) << ", "
+			<< domainDecomp->getBoundingBoxMax(2, domain) << endl;
 #endif
 	std::vector< std::vector<Molecule> > binVector;
 	binVector.resize(numBins);
 	uint32_t nBinIndex;
-	for(auto&& mol:_particleVector)
+	for(auto mol:_particleVector)  // auto&&
```

Comment:
a) what does this comment mean?
b) you are sure you want copies of the molecules here? (would make sense to state your intentions in a comment here)

Reply:
I have to admit that Im not really sure. If I remember right, it did not work when using "auto&&". Perhabs I should check that again.

Reply:
I've checked it now. Need to copy Molecule objects here. But perhaps handling of reservoir Molecule could be revised in general, to avoid or at least to reduce copy events of Molecule objects.